Automattic/vip-go-mu-plugins
# Disable Contact Form 7 upload cleanup if it ignores WPCF7_UPLOADS_TMP…

## Description
TL;DR this stops Contact Form 7 attempting to open a directory within wp-content, which raises excessive `opendir() Failed to open directory` warnings.

This PR addresses a situation where recent versions of the Contact Form 7 (CF7) enforces a policy of `WPCF7_UPLOADS_TMP_DIR` being located somewhere within `wp-content`, which isn't compatible with VIP platform. CF7 has a method `wpcf7_upload_tmp_dir()` which uses the value of `WPCF7_UPLOADS_TMP_DIR` but expects it to be within the `wp-content` directory, and ultimately returns a path within `wp-content/uploads/` on VIP Platform. 

The current version of the plugin (5.7.2 at time of writing) does not currently work for attachments, without patching - this PR _does not_ address that.

The reason for this PR is that a method run on shutdown of every request attempts to run a method `wpcf7_cleanup_upload_files()`, which tries to run `opendir()` in a sub-folder of `wp-content`. This generates excessive `opendir() Failed to open directory` warnings.

## Changelog Description

### Plugin fixes updated to include a workaround for Contact Form 7 upload file cleanup

An action was added to conditionally disable the `wpcf7_cleanup_upload_files` shutdown action, if the tmp directory path returned by `wpcf7_upload_tmp_dir()` does not match the value in `WPCF7_UPLOADS_TMP_DIR`.

## Checklist


## Steps to Test

1. Install a recent Contact Form 7 (e.g. 5.7.2) on a VIP test site
1. Visit the test site and request some pages
1. Check PHP logs for `opendir() Failed to open directory` warnings.
1. Checkout this PR and switch the site to use the appropriate version of mu-plugins
1. Visit the test site and request some pages
1. Check that there are no longer`opendir() Failed to open directory` warnings being triggered.
